### DISCUSSION TITLE
Running Brickstore from QT Creator

### DISCUSSION THREAD

**Discussion by ErfinderDesRades** [rgriebl/brickstore#550] [General]
Hello everybody,

I'm trying to run Brickstore from QT-Creator without success. I need to know, whether I used the right source to download the source code.
So my question is: When I download the zip file from this url : https://github.com/rgriebl/brickstore, do I have all that's necessary to run Brickstore from QT-Creator or do I need anything else?

Thanks a lot in advance for any response

Regards
Andreas

**Comment by rgriebl** [comment]
I've just updated https://github.com/rgriebl/brickstore/wiki/Building-from-Source to be a bit more specific when it comes to Windows (which my crystal ball is telling me you might be using ;) )

That being said: "without success" isn't a known error message. If you want help, you need to post the error messages you are getting.

And yes - each release at https://github.com/rgriebl/brickstore/releases comes with a ZIP + Tarball containing the sources of that release.

**Comment by ErfinderDesRades** [comment]
First of all thanks for your response and sorry for expressing myself in a maybe cryptic way. It wasn't my intention to post a riddle. I first of all wanted to make sure, that I used the right source code.
I'm indeed using Windows 10 and, after reading your response I deinstalled the QT package completely an reinstalled in exactly the way you described in the "Wiki Building from source".
After this I opened the brickstore project and tried to setup a build. When following your advice to use an MSCV kit I first get the information that there is no C++ compiler in this kit. So I just choosed a MING compiler as no other one has been offered (please refer to  following screenshot) 

<img width="977" alt="QTCreator_01" src="https://user-images.githubusercontent.com/77791504/168563060-dbcd8431-a944-41e4-9e31-886f5d27eb38.png">

After running the build I received 67 error messages. 66 of those were concerning missing version numbers in the import declarations of the qml files, ,for example like this one:

<img width="305" alt="QTCreator_02" src="https://user-images.githubusercontent.com/77791504/168564174-6d21714d-8363-4c68-9bc7-e9bc3c409caf.png">

inside the code it was written like this:

<img width="732" alt="QTCreator_03" src="https://user-images.githubusercontent.com/77791504/168564263-52993dfc-0342-4b4e-8295-db1823c355f2.png">

Error message number 67 was this one:

<img width="577" alt="QTCreator_04" src="https://user-images.githubusercontent.com/77791504/168564512-0741d093-9ae3-40c7-ab3f-b21fcc3558ac.png">

Obviously concerning the used kit.

Moreover there was one compiler message like this:

<img width="887" alt="QTCreator_05" src="https://user-images.githubusercontent.com/77791504/168564981-ec63f036-e7d3-471b-9543-0cf7d381cc5d.png">

Just for explanation: OpenSSL and ImmoSetup are installed.

So in the next step I used another kit, this time a MinGW 64 bit (yes I know you recommended to use MSCV kits, but those did not work):

<img width="979" alt="QTCreator_06" src="https://user-images.githubusercontent.com/77791504/168566675-994b0743-4812-4e50-b7c7-0cd838d213d4.png">

After doing the build error message number 67 (concerning the kit) was gone, but the 66 messages concerning the version numbers remained and the compiler message changed to this:

<img width="887" alt="QTCreator_07" src="https://user-images.githubusercontent.com/77791504/168567399-29cccaf3-22a3-4eab-9133-cc47875da7d9.png">

So in the next step I added the requested version numbers (I choosed 2.12) for all the Quick imports and started the next build, again using the MinGW kit.
The 66 messages concerning the version numbers disappeared, the compiler message changed to:
 
<img width="960" alt="QTCreator_08" src="https://user-images.githubusercontent.com/77791504/168574203-d391da66-31b5-4dce-a6e7-618e686ec341.png">

That's the current status, because right now I don't know how to go on.

Sorry for that huge post, but I didn't know another way to explain what my problem is. I'm sure the problem is sitting in font of the computer, but I don't know how to solve it.

Thanks a lot in advance for dealing with this issue.

Regards from sunny Bremen
Andreas

**Reply by rgriebl** [reply]
Ignore all the "missing import version" warnings ... this is for the mobile (Android/iOS) UI variant, which requires Qt6.
These won't affect you -- it's just QtCreator being overzealous :)

The only real error is that the build cannot find your OpenSSL libraries.
Are you sure you installed `Win64 OpenSSL v1.1.1o Light` from https://slproweb.com/products/Win32OpenSSL.html ?
If you really did, then you have to adjust `brickstore.pro` around line 200 to make it find your installation. (or make sure that your OpenSSL installation is actually ending up in one of the well-known locations that brickstore uses to search for the libs)

**Reply by ErfinderDesRades** [reply]
Ok thanks a lot for your hint. Right now I'm on a business trip and will be back on Friday. As soon as I'll be back I will check the path of OpenSSL and brickstore.pro.

Have a nice evening

**Comment by ErfinderDesRades** [comment]
I followed your hint and adjusted the path at line 214 in brickstore.pro. After this the error message concerning OpenSSL disappeared (yeah!!!).
But now there are three different error messages:

<img width="382" alt="QTCreator_09" src="https://user-images.githubusercontent.com/77791504/169813317-54194316-f26f-4fe9-804e-a9be8a69b7d6.png">

<img width="385" alt="QTCreator_10" src="https://user-images.githubusercontent.com/77791504/169813351-f5146ecd-950d-4f8e-bd09-43f09cb94771.png">

as far as I can see these two are concerning the file qcoro.pri:

<img width="677" alt="QTCreator_11" src="https://user-images.githubusercontent.com/77791504/169813545-7df4832b-f582-4536-bc72-5e348018f13f.png">

During the weekend I read a lot documentations for QT and qmake. I also checked the command line options for qmake, but couldn't find out what to do to solve this problem.

The third problem I couldn't solve comes with this message:

<img width="249" alt="QTCreator_12" src="https://user-images.githubusercontent.com/77791504/169814012-91850710-730d-4c48-9631-5d13ff8f8f00.png">

Sorry for maybe bothering you, but now that I tasted blood, I really want to go on and see whether I can build some extensions by myself.

Kind regards

**Reply by rgriebl** [reply]
As mentioned in https://github.com/rgriebl/brickstore/wiki/Building-from-Source

> MingW on Windows is untested - please use the compiler from MSVC 2019 or 2022. If you still want to use MingW, be my guest and create a pull request if you get it running!

And given your error message, it is clear that your MingW version is too old to even try to get it working (it lacks support for C++20 co-routines).
Just install the MSVC 2019 community edition from here https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes

**Comment by ErfinderDesRades** [comment]
Directly after posting my last comment, I tried the kit MSVC2019 64bit which I installed together with the QT version 6.2.4 and this one worked perfectly. Really Great.

Thank you so much for your long-suffering support. Now I will finish my studies about QT IDE and then I can start to check out the brickstore source code.

Kind regards from Bremen
